From 2294e61b057215bf089b1d1c482a5dcbcad6b414 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Thu, 23 May 2024 17:50:18 +0200
Subject: [PATCH] src/platform.hh: fix or1k build

Fix the following or1k build failure raised since version 0.2.2 and
https://github.com/q66/cffi-lua/commit/27755f61e4c625dff951302bade4ad07e3609ca5:

In file included from /home/buildroot/autobuild/run/instance-0/output-1/host/or1k-buildroot-linux-uclibc/sysroot/usr/include/lua.hpp:6,
                 from ../src/lua.hh:23,
                 from ../src/parser.hh:4,
                 from ../src/ffilib.cc:7:
../src/ffilib.cc: In static member function 'static void ffi_module::setup(lua_State*)':
../src/ffilib.cc:1616:28: error: expected ')' before 'FFI_ARCH_NAME'
 1616 |         lua_pushliteral(L, FFI_ARCH_NAME);
      |                            ^~~~~~~~~~~~~

Fixes: 27755f61e4c625dff951302bade4ad07e3609ca5
 - http://autobuild.buildroot.org/results/4e14753732c5b6fe8ba9ecc4050ffb35f471c428

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Upstream: https://github.com/q66/cffi-lua/commit/2294e61b057215bf089b1d1c482a5dcbcad6b414
---
 src/platform.hh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/platform.hh b/src/platform.hh
index 84e0a89..dab1e76 100644
--- a/src/platform.hh
+++ b/src/platform.hh
@@ -192,7 +192,7 @@
 #  endif
 #elif defined(__OR1K__)
 #  define FFI_ARCH FFI_ARCH_OR1K
-#  define FFO_ARCH_NAME "or1k"
+#  define FFI_ARCH_NAME "or1k"
 #elif defined(__riscv) || defined(__riscv__)
 #  if __riscv_xlen == 32
 #    define FFI_ARCH FFI_ARCH_RV32
